app/scripts/metamask-controller.js

```diff
@@ -2456,18 +2459,21 @@ export default class MetamaskController extends EventEmitter {
    * array if no accounts are permitted.
    *
    * @param {string} origin - The origin whose exposed accounts to retrieve.
+   * @param {boolean} [suppressUnauthorizedError] - Suppresses the unauthorized error.
    * @returns {Promise<string[]>} The origin's permitted accounts, or an empty
    * array.
    */
-  async getPermittedAccounts(origin) {
+  async getPermittedAccounts(origin, suppressUnauthorizedError = true) {
     try {
       return await this.permissionController.executeRestrictedMethod(
         origin,
         RestrictedMethods.eth_accounts,
       );
     } catch (error) {
-      if (error.code === rpcErrorCodes.provider.unauthorized) {
-        return [];
+      if (suppressUnauthorizedError) {
+        if (error.code === rpcErrorCodes.provider.unauthorized) {
```

Comment:
These can be combined, no? `if suppressUnauthorizedError && error.code === rpcErrorCodes.provider.unauthorized`.